GVFS/GVFS.Platform.Windows/WindowsFileSystemVirtualizer.cs

```diff
@@ -1046,6 +1034,32 @@ private HResult QueryFileNameHandler(string virtualPath)
             }
         }
 
+        private void MarkDirectoryAsPlaceholder(
+            string virtualPath,
+            uint triggeringProcessId,
+            string triggeringProcessImageFileName)
+        {
+            string directoryPath = Path.Combine(this.Context.Enlistment.WorkingDirectoryRoot, virtualPath);
+            HResult hr = this.virtualizationInstance.MarkDirectoryAsPlaceholder(
+                directoryPath,
+                FolderContentId,
+                PlaceholderVersionId);
+
+            if (hr == HResult.Ok)
+            {
+                this.FileSystemCallbacks.OnPlaceholderFolderCreated(virtualPath, triggeringProcessImageFileName);
+            }
+            else
+            {
+                EventMetadata metadata = this.CreateEventMetadata(virtualPath);
+                metadata.Add("isDirectory", true);
+                metadata.Add("triggeringProcessId", triggeringProcessId);
+                metadata.Add("triggeringProcessImageFileName", triggeringProcessImageFileName);
+                metadata.Add("HResult", hr.ToString());
+                this.Context.Tracer.RelatedError(metadata, nameof(this.NotifyNewFileCreatedHandler) + "_" + nameof(this.virtualizationInstance.MarkDirectoryAsPlaceholder) + " error");

```

Comment:
I think you could probably replace 

`nameof(this.NotifyNewFileCreatedHandler) + "_" + nameof(this.virtualizationInstance.MarkDirectoryAsPlaceholder)` 

with 

`nameof(this.MarkDirectoryAsPlaceholder)`